### PR TITLE
Deprecate DirectDiscretizationConnection.full_resample_matrix

### PR DESCRIPTION
**Pull request by alexfikl**
Fixes https://gitlab.tiker.net/inducer/meshmode/-/issues/13

### CODE REVIEW

File: meshmode/discretization/connection/direct.py [hunk 1]
```diff
@@ -412,4 +348,84 @@ def pick_knl():
 # }}}
 
 
+# {{{ dense resampling matrix
+
+def make_direct_full_resample_matrix(actx, conn):
```
Comment (inducer) [inline meshmode/discretization/connection/direct.py:353]: Did you mean to leave this out of the module docstring?

Comment (alexfikl) [reply meshmode/discretization/connection/direct.py:353]: Kind of, given all the warnings for its usage. I can add it back though, if you want.

Comment (inducer) [reply meshmode/discretization/connection/direct.py:353]: No, that's OK. Just wanted to make sure it was intentional.

**Comment by inducer** [comment]
Thanks!

**Pull request merged**